libs/libarchfpga/src/read_xml_arch_file.cpp

```diff
@@ -4791,6 +4795,16 @@ static void link_physical_logical_types(std::vector<t_physical_tile_type>& Physi
             archfpga_throw(__FILE__, __LINE__,
                            "Logical Block %s does not have any equivalent tiles.\n", logical_block.name);
         }
+
+        std::unordered_map<int, bool> ingored_pins_check_map;
+        std::unordered_map<int, bool> global_pins_check_map;
+
+        for (int pin = 0; pin < logical_block.pb_type->num_pins; pin++) {
+            for (auto& tile : logical_block.equivalent_tiles) {
+                auto direct_map = tile.tile_block_pin_directs_map[logical_block.index];
+                int phy_index = direct_map[t_logical_pin(pin)].pin;
```

Comment:
This should probably be `direct_map.at(t_logical_pin(pin))` to add error checking if the logical pin is missing from the map.

Reply:
Done